### ISSUE TITLE
style-loader unable to work with both less and sass

### ISSUE THREAD

**Issue by Ognian** [neutrinojs/neutrino#1561]
"@neutrinojs/react": "^9.1.0",
.neutrinorc.js
```
const react = require('@neutrinojs/react');
const jest = require('@neutrinojs/jest');
const path = require('path');

module.exports = {
  options: {
    root: __dirname,
  },
  use: [
    react({
      html: {
        title: 'imsquerybuilder'
      },
      style:{
        loaders: [
          {
            loader: 'sass-loader',
            useId: 'sass'
          },
          {
            loader: 'less-loader',
            useId: 'less',
            options: {
              javascriptEnabled: true,
            }
          },
        ],
        test: /\.(css|less|sass|scss)$/,
        modulesTest: /\.module.(css|less|sass|scss)$/,
      }
      }
    ),
    jest(),
  ],
};
```

I need to use both less and sass. The above config leads to errors.
```
ERROR in ./node_modules/react-awesome-query-builder/css/compact_styles.scss (./node_modules/css-loader/dist/cjs.js??ref--6-oneOf-1-1!./node_modules/sass-loader/dist/cjs.js!./node_modules/less-loader/dist/cjs.js??ref--6-oneOf-1-3!./node_modules/react-awesome-query-builder/css/compact_styles.scss)
Module build failed (from ./node_modules/less-loader/dist/cjs.js):

```

If I comment out less in the above config and use only sass files it works.
If I comment out sass in the above config and use only less files it works too.
So only if I need both it doesn't work.
Looks like loaders are chained which looks wrong for me
how would I configure loaders with their own `tests`and `modulesTest` in the style option of react?
Thanks
Ognian

**Comment by edmorley** [comment]
@Ognian Hi!

The chaining of loaders using that configuration is by design, since there are most of the CSS related loaders need to be combined with other loaders in order to work. (There are quite possibly changes that can be made to the docs to make this clearer though.)

For example when using sass, webpack can't just use `sass-loader`, since that only converts the sass files to css (and not JS, which is all webpack understands), and won't either inline the CSS (in the case of development) or output the CSS files separately (in the case of production).

The actual chain of loaders for some users could end up being as complex as:
`sass-loader -> postcss-loader -> css-loader -> {style-loader or mini-css-extract-plugin}`

It sounds like in your use-case where you're using both less and sass, you'll need multiple style rules configured (both of which have their own chained loaders).

In which case it would end up looking like:
`sass-loader -> css-loader -> {style-loader or mini-css-extract-plugin}`
plus in parallel:
`less-loader -> css-loader -> {style-loader or mini-css-extract-plugin}`

Now `@neutrinojs/style` (which is what `@neutrinojs/react` uses internally for the `style` config) handles the `css-loader` and `{style-loader or mini-css-extract-plugin}` parts for you, but this still means you need to use multiple instances of `@neutrinojs/style`, and not just the one default `@neutrinojs/style` instance configured by `@neutrinojs/react`.

Ideally `@neutrinojs/react` (and other web presets) would support passing an array to `style` (and not just an object) such that they create multple instances of `@neutrinojs/style` for you. There is a feature request open for this:
https://github.com/neutrinojs/neutrino/issues/964

In the meantime I would do something like:

```js
const react = require('@neutrinojs/react');
const style = require('@neutrinojs/style');
//...

module.exports = {
  //...
  use: [
    react({
      // ...
      style:{
        loaders: [
          {
            loader: 'sass-loader',
            useId: 'sass'
          },
        ],
        test: /\.(css|sass|scss)$/,
        modulesTest: /\.module.(css|sass|scss)$/,
      }
    }),
    // Add a second instance of `@neutrinojs/style` for less support
    style({
      ruleId: 'style-less',
      loaders: [
        {
          loader: 'less-loader',
          useId: 'less',
          options: {
            javascriptEnabled: true,
          }
        },
      ],
      test: /\.(less)$/,
      modulesTest: /\.module.(less)$/,
    }),
    //...
  ],
};
```

I haven't tested this locally, but should hopefully give an idea of how to do this. The docs for configuring `@neutrinojs/style` are here:
https://neutrinojs.org/packages/style-loader/

I would also take advantage of `neutrino --inspect` (to see the webpack config Neutrino generates) to assist with debugging:
https://neutrinojs.org/usage/#inspecting-the-generated-webpack-config

The webpack docs on loaders may also be useful:
https://webpack.js.org/concepts/loaders/

Good luck! :-)

**Comment by edmorley** [comment]
Duplicate of #964

**Comment by constgen** [comment]
Have implemented Sass and Less middlewares recently that excellently integrate with core `style-loader` 
- https://github.com/constgen/neutrino-middleware-sass-loader
- https://github.com/constgen/neutrino-middleware-less-loader

**Comment by bioinformatist** [comment]
@edmorley I would really appreciate to you for the reply to this issue, which also saves tons of my time. Best wishes!
